### PR TITLE
Add API docs

### PR DESCRIPTION
**Pull request by thunderbiscuit**
## Description
This PR adds the required files and plugin to generate API docs using [Dokka](https://github.com/Kotlin/dokka).

## Notes to the reviewers
While the process requires some manual input from a maintainer before release, I believe it is now down to manageable state. The process requires the maintainer to apply a patch file to the `bdk.kt` bindings file, and manually fixing up the hunks that the patch tool cannot automatically resolve, as well as adding new KDocs if anything has been added to the new release. The process requires a few steps but is clearly defined.

I'm not sure where I should officially keep this process explained, but for now it's in #18.

I have done this process 3 times now, learning about the `patch` and `diff` commands along the way (picking the right arguments for a successful patch is somewhat of an art). I think I have something that works fairly well, at least for small increments in the API.

## All submissions:
* [x] I've signed all my commits

### CODE REVIEW

**Comment by thunderbiscuit** [comment]
At the moment, only minimal things are added and most of the docstrings are simply suppressing constructs that are built by the bindings library and don't need to appear in the KDocs. 

Once we agree on the approach and this gets merged, I'll spend time to actually bring in the full extent of the docs that are in the Rust bdk project.

File: android/Module.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Module bdk-android
+The [bitcoindevkit](https://bitcoindevkit.org/) language bindings library for Android. Current version: `0.5.1`.
```
Comment (notmandatory) [inline android/Module.md:2]: Is it customary to put the latest version in this file? If not I'd rather leave it out so we only need to bump versions in the `build.gradle.kts` file.

```suggestion
The [bitcoindevkit](https://bitcoindevkit.org/) language bindings library for Android.
```

File: jvm/Module.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Module bdk-jvm
+The [bitcoindevkit](https://bitcoindevkit.org/) language bindings library for the JVM. Current version: `0.5.1`.
```
Comment (notmandatory) [inline jvm/Module.md:2]: Same comment as for `android/Module.md`.

```suggestion
The [bitcoindevkit](https://bitcoindevkit.org/) language bindings library for the JVM.
```

File: android/Module.md (hunk 1 above)
Comment (thunderbiscuit) [reply android/Module.md:2]: No I don't think it is! But I haven't found how to display the version so far. So the docs are ambiguous as to which version they document... I'm sure there is a simple way but for some reason I couldn't find it.

**Comment by notmandatory** [comment]
Tested and works great for me! I think it's OK to leave the steps to build in #18 until we figure out if/how it can be automated in `build.sh`.

File: jvm/Module.md (hunk 2 above)
Comment (thunderbiscuit) [reply jvm/Module.md]: Done.

File: android/Module.md (hunk 1 above)
Comment (thunderbiscuit) [reply android/Module.md]: Done.

**Review by notmandatory — APPROVED** [review]
ACK 42b8db8609a01fe79bf602c85f3d7c8b915be520

**Pull request merged**